src/app/home/event-user-list/event-user-list.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input, ViewChild, Output, EventEmitter } from '@angular/core';
+import { ClrModal } from '@clr/angular';
+import { Progress } from 'src/app/data/progress';
+import { User } from 'src/app/data/user';
+
+interface userListUser extends User {
+  numberOfSessions?: number;
+  activeScenarioName?: string;
+}
+
+@Component({
+  selector: 'event-user-list',
+  templateUrl: './event-user-list.component.html'  
+})
+export class EventUserListComponent implements OnInit {
+
+  @ViewChild("userListModal") userListModal: ClrModal;
+
+  @Input()
+  public users: User[];
+
+  @Input()
+  public progress: Progress[]; 
+
+  @Input()
+  public filterName: Function;
+
+  @Output() nameClickedEvent = new EventEmitter<string>();
+  
+  public userListUsers: userListUser[] = [];
+
+  ngOnInit(): void {}
+
+  public listFilterName(username) {
+    this.nameClickedEvent.emit(username);
+    this.userListModal.close()
+  }
+
+  public openModal(): void {
+    this.userListUsers = []
+    this.users.forEach(user => {
+      let newUser: userListUser = user;
+      newUser.activeScenarioName = this.getUsersActiveProgressName(user);
+      newUser.numberOfSessions = this.getUsersProgress(user);
+      this.userListUsers.push(newUser)
+    })
```

Comment:
This is a perfect use-case for `map` instead of `forEach`:

```suggestion
    this.userListUsers = this.users.map(user => ({
      ...user,
      activeScenarioName: this.getUsersActiveProgressName(user),
      numberOfSessions: this.getUsersProgress(user),
    }))
```

With this purely functional approach, you can also make the two new fields of `UserListUser` required instead of optional. Moreover, you do not mutate the existing user objects, which could have unexpected side-effects.